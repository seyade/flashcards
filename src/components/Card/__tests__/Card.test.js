import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Card } from '../index';

configure({ adapter: new Adapter() });

const props = {
  card: {
    id: 0,
    prompt: 'test prompt',
    answer: 'test answer',
  },
};

describe('Card', () => {
  const card = shallow(<Card {...props} />);

  it('renders correctly', () => {
    const tree = renderer.create(card).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sets `reveal` to be `false`', () => {
    expect(card.state().reveal).toBe(false);
  });

  it('renders to card prompt', () => {
    expect(card.find('.card__prompt h4').text()).toEqual(props.card.prompt);
  });

  it('renders to card answer', () => {
    expect(card.find('.card__answer h4').text()).toEqual(props.card.answer);
  });

  it('applies the `text-hidden` class to the card answer', () => {
    expect(card.find('.card__answer h4').hasClass('text-hidden')).toBe(true);
  });

  describe('when clicking on the card', () => {
    beforeEach(() => {
      card.simulate('click');
    });

    it('updates `reveal` to be `true`', () => {
      expect(card.state().reveal).toBe(true);
    });

    it('applies the `text-revealed` class to the card answer', () => {
      expect(card.find('.card__answer h4').hasClass('text-revealed')).toBe(
        true
      );
    });
  });
});
