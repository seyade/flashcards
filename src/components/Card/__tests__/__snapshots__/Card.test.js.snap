// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<div
  className="sc-bdVaJa gkVvnR"
  onClick={[Function]}
>
  <div
    className="card__header"
  >
    <div
      className="card__prompt"
    >
      <h4>
        test prompt
      </h4>
    </div>
  </div>
  <div
    className="card__content"
  >
    <div
      className="card__answer"
    >
      <h4
        className="text-hidden"
      >
        test answer
      </h4>
    </div>
  </div>
</div>
`;
